app/database/schema.js

```diff
@@ -220,6 +221,10 @@ const schemaModel = {
 			})
 		)[0]?.id;
 
+		if (existingService && !schemaId && await versionExists(trx, existingService.id, service.version)) {
+			throw new PublicError(`Schema [${existingService.name}] and version [${service.version}] already exist in registry`);
```

Comment:
I first didn't understand how this if can happen (re-read the title/description, so its clear now), maybe add a hint in the message, ex. `you should not register different type_defs with same version`?

Reply:
Sounds like a good idea!